library/promptmanager.py

```diff
@@ -51,26 +54,28 @@ def get_latest_prompt_template(self, name: str):
             raise PromptMissingException(f"No template found for prompt `{name}`: {prompt}")
         return template
 
-    def get_prompt(self, name: str):
-        prompt_id = self.name_to_id(name)
-        result = self.get_response("get-prompt", {"prompt_id": prompt_id})
+    def get_prompt(self, name: str) -> dict[str, str]:
+        prompt_id: str = self.name_to_id(name)
+        result: dict[str, str] = self.get_response("get-prompt", {"prompt_id": prompt_id})
         return result
 
-    def cache_prompt_keys(self):
-        prompts = self.get_prompts()
+    def cache_prompt_keys(self) -> None:
+        prompts: dict[str, any] = self.get_prompts()
         self.prompt_cache = {prompt["name"]: prompt["id"] for prompt in prompts}
-        self.cache_complete = datetime.datetime.now()
+        self.cache_complete: datetime.datetime = datetime.datetime.now()
         print("Prompt cache completed: ", self.prompt_cache)
 
-    def get_prompts(self):
+    def get_prompts(self) -> dict[str, any]:
         result = self.get_response("get-prompts", {})
         return result
 
-
-    def get_response(self, request: str, params: dict) -> dict:
-        url = self.base_url + request
-        params.update({"branch_id": self.branch_id, "api-key": self.api_key})
-        response = requests.get(url, headers=self.headers, params=params)
+    @d.deep_freeze_args
```

Comment:
This is using the "deep_freeze_args" method to accommodate the fact that `params` is a `dict[.,.]`. It does the transformation before the caching then caches with a TTL of 1 day.

The code in the method then changes to accommodate the fact that the `params` is now a "frozen dict" and can't be modified (eg. the `update` method). 

Looking at this it makes me think perhaps there should be a more comprehensive caching decorator that wraps both of these and ensures that thee original value is passed to the method of the cache hit fails.

Reply:
I'm a bit confused here. You cached request and params using the decorator: deep_freeze_args and then updated params?

Reply:
Not quite. I used a library (`cachetools` and `@cached`) to cache the thing based on parameters sent to the method. That library (unsurprisingly) used a hash to represent the state of what's sent. So I had to do something to deal with the fact a mutable object like a `dict` cannot be hashed, and I leveraged another decorator to do it - which I wrote myself based on discussions on stack overflow.

Reply:
Okay, I get it. So the mutable dict is transformed into a frozen_dict after it is updated, allowing it to be cached. If there's an issue retrieving from the cache using the frozen_dict, the original value can be used to retrieve the information from the cache.

However, this approach would only work correctly if the original value doesn't change over time and remains the same as the frozen_dict that was initially cached.